dts/arm/nxp/nxp_lpc11u6x.dtsi

```diff
@@ -52,6 +52,32 @@
 			status = "okay";
 		};
 
+		iocon: iocon@40044000 {
```

Comment:
why not name this one `pinctrl`?

Reply:
The `pio0` through `pio2` nodes are used by the GPIO driver to determine base addresses for each IOCON port, so the `compatible` string for the `iocon` node needs to define the properties for each `pio` child node. Then, the child `pinctrl` node has its own properties, and that node has the pincontrol selection defined as children.

I believe if this node was `pinctrl` then the `pio` child nodes could not be present, because they do not contain the same properties as pin control groups (which are the children of the `pinctrl` node) do. Please correct me if that is wrong, because this node being named `pinctrl` would be cleaner.

Reply:
If a child has a compatible set, then it can do whatever it wants, see for example STM32/GD32 pinctrl/gpio layout.

Reply:
Makes sense. The only other hurdle to renaming that pin control node would be that the lpc pin control driver currently relies on the node label `iocon` being present in the devicetree in order to get the base address of the pin controller. I used the node label directly because like the iMX pin control, there are several different register layouts present across the SOC families with the IOCON peripheral, so there is not one shared DTS compatible string that could be used for all pin control nodes.

So I could either:
- rename the `iocon` node for all devices currently using this pin control driver (and rework the dts binding file (currently the RT600 and all LPC SOCs use this driver)
- add a corner case to this driver for the LPC11u6x

personally, I would rather keep the driver and node naming as it currently is designed, but if you feel strongly about this, I can rename the `iocon` node across the LPC line. I would prefer not to add a corner case for the LPC11u6x, however.

Reply:
I think it would make more sense to make `iocon` the pinctrl (following stm32/gd32 layout), but this can be improved later.